<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Mid-Cayman Rise analysis - 18S survey</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Mid-Cayman Rise analysis</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./grazing.html">Grazing analysis</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./asv.html" aria-current="page">Sequence analysis</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#compile-asv-data-from-mcr-from-previous-sequence-analysis" id="toc-compile-asv-data-from-mcr-from-previous-sequence-analysis" class="nav-link active" data-scroll-target="#compile-asv-data-from-mcr-from-previous-sequence-analysis">Compile ASV data from MCR from previous sequence analysis</a></li>
  <li><a href="#import-metadata" id="toc-import-metadata" class="nav-link" data-scroll-target="#import-metadata">Import metadata</a>
  <ul class="collapse">
  <li><a href="#import-decontam-results-to-qc-asv-table" id="toc-import-decontam-results-to-qc-asv-table" class="nav-link" data-scroll-target="#import-decontam-results-to-qc-asv-table">Import decontam results to QC ASV table</a></li>
  <li><a href="#import-as-phyloseq-objects" id="toc-import-as-phyloseq-objects" class="nav-link" data-scroll-target="#import-as-phyloseq-objects">Import as phyloseq objects</a></li>
  <li><a href="#save-output-asv-tables-for-downstream-analyses" id="toc-save-output-asv-tables-for-downstream-analyses" class="nav-link" data-scroll-target="#save-output-asv-tables-for-downstream-analyses">Save output ASV tables for downstream analyses</a></li>
  <li><a href="#sequence-stats" id="toc-sequence-stats" class="nav-link" data-scroll-target="#sequence-stats">Sequence stats</a></li>
  </ul></li>
  <li><a href="#visualize-metadata" id="toc-visualize-metadata" class="nav-link" data-scroll-target="#visualize-metadata">Visualize metadata</a></li>
  <li><a href="#community-structure-diversity-at-von-damm-piccard" id="toc-community-structure-diversity-at-von-damm-piccard" class="nav-link" data-scroll-target="#community-structure-diversity-at-von-damm-piccard">Community structure &amp; diversity at Von Damm &amp; Piccard</a>
  <ul class="collapse">
  <li><a href="#cluster-analysis" id="toc-cluster-analysis" class="nav-link" data-scroll-target="#cluster-analysis">Cluster analysis</a></li>
  <li><a href="#cluster-analysis-with-grazing-assays" id="toc-cluster-analysis-with-grazing-assays" class="nav-link" data-scroll-target="#cluster-analysis-with-grazing-assays">Cluster analysis with grazing assays</a>
  <ul class="collapse">
  <li><a href="#pcoa-with-all-samples" id="toc-pcoa-with-all-samples" class="nav-link" data-scroll-target="#pcoa-with-all-samples">PCoA with all samples</a></li>
  <li><a href="#pcoa-with-in-situ-only" id="toc-pcoa-with-in-situ-only" class="nav-link" data-scroll-target="#pcoa-with-in-situ-only">PCoA with in situ only</a></li>
  </ul></li>
  <li><a href="#shared-asvs" id="toc-shared-asvs" class="nav-link" data-scroll-target="#shared-asvs">Shared ASVs</a></li>
  <li><a href="#richness-by-site" id="toc-richness-by-site" class="nav-link" data-scroll-target="#richness-by-site">Richness by site</a></li>
  <li><a href="#corncob" id="toc-corncob" class="nav-link" data-scroll-target="#corncob">Corncob</a></li>
  </ul></li>
  <li><a href="#compare-in-situ-and-tf-samples" id="toc-compare-in-situ-and-tf-samples" class="nav-link" data-scroll-target="#compare-in-situ-and-tf-samples">Compare in situ and Tf samples</a></li>
  <li><a href="#session-information" id="toc-session-information" class="nav-link" data-scroll-target="#session-information">Session information</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">18S survey</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="compile-asv-data-from-mcr-from-previous-sequence-analysis" class="level1">
<h1>Compile ASV data from MCR from previous sequence analysis</h1>
<p>Set up R environment</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(phyloseq); <span class="fu">library</span>(ape)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Import previously sequenced and analyzed tag-sequence data. See https://shu251.github.io/microeuk-amplicon-survey/ for additional information.</p>
</section>
<section id="import-metadata" class="level1">
<h1>Import metadata</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../microeuks_deepbiosphere_datamine/microeuk-amplicon-survey/data-input/samplelist-metadata.csv"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># head(metadata)</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(metadata$Sample_actual)</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>mcr <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"VonDamm"</span>, <span class="st">"Piccard"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>metadata_formatted <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SITE <span class="sc">%in%</span> mcr) <span class="sc">%&gt;%</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(as.character) <span class="sc">%&gt;%</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample_or_Control <span class="sc">==</span> <span class="st">"Sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(SAMPLETYPE <span class="sc">==</span> <span class="st">"Microcolonizer"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SAMPLE, VENT, SITE, SAMPLEID, DEPTH, SAMPLETYPE, YEAR, <span class="at">TEMP =</span> <span class="fu">starts_with</span>(<span class="st">"TEMP"</span>), pH, <span class="at">PercSeawater =</span> <span class="fu">starts_with</span>(<span class="st">"Perc"</span>), <span class="at">Mg =</span> <span class="fu">starts_with</span>(<span class="st">"Mg"</span>), <span class="at">H2 =</span> <span class="fu">starts_with</span>(<span class="st">"H2."</span>), <span class="at">H2S =</span> <span class="fu">starts_with</span>(<span class="st">"H2S"</span>), <span class="at">CH4 =</span> <span class="fu">starts_with</span>(<span class="st">"CH4"</span>), ProkConc, Sample_or_Control)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="import-decontam-results-to-qc-asv-table" class="level3">
<h3 class="anchored" data-anchor-id="import-decontam-results-to-qc-asv-table">Import decontam results to QC ASV table</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"../../microeuks_deepbiosphere_datamine/microeuk-amplicon-survey/seq-analysis/contam-asvs.RData"</span>, <span class="at">verbose=</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  list_of_contam_asvs</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># class(list_of_contam_asvs)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>asv_wtax_qc <span class="ot">&lt;-</span> merged_asv <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">FeatureID =</span> <span class="st">'#OTU ID'</span>, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(FeatureID <span class="sc">%in%</span> list_of_contam_asvs)) <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># In wide format, subsample 1000 random ASVs</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># sample_n(1000, replace = FALSE) %&gt;%</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">!</span>FeatureID,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"SAMPLE"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"_MCR_"</span>, SAMPLE)) <span class="sc">%&gt;%</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(merged_tax, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"FeatureID"</span> <span class="ot">=</span> <span class="st">"Feature ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">filter</span>(metadata_formatted, <span class="fu">grepl</span>(<span class="st">"_MCR_"</span>, SAMPLE))) <span class="sc">%&gt;%</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(SAMPLENAME, SITE, SAMPLETYPE, YEAR, VENT, SAMPLEID, <span class="at">sep =</span> <span class="st">" "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(asv_wtax_qc)</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(asv_wtax_qc$FeatureID))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>tax_matrix <span class="ot">&lt;-</span> asv_wtax_qc <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, Taxon) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Taxon, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Supergroup"</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  as.matrix</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(tax_matrix)</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>asv_matrix <span class="ot">&lt;-</span> asv_wtax_qc <span class="sc">%&gt;%</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, SAMPLE, value) <span class="sc">%&gt;%</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SAMPLE, <span class="at">values_from =</span> value, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  as.matrix</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(asv_matrix)</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(asv_matrix); dim(tax_matrix)</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Align row names for each matrix</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tax_matrix) <span class="ot">&lt;-</span> <span class="fu">row.names</span>(asv_matrix)</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(asv_matrix)</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>mcr_samples <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">colnames</span>(asv_matrix))</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Set rownames of metadata table to SAMPLE information</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>metadata_mcr <span class="ot">&lt;-</span> <span class="fu">filter</span>(metadata_formatted, SAMPLE <span class="sc">%in%</span> mcr_samples) <span class="sc">%&gt;%</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"X"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"SAMPLE"</span>)</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(metadata_mcr)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="import-as-phyloseq-objects" class="level3">
<h3 class="anchored" data-anchor-id="import-as-phyloseq-objects">Import as phyloseq objects</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import asv and tax matrices</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">=</span> <span class="fu">otu_table</span>(asv_matrix, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>TAX <span class="ot">=</span> <span class="fu">tax_table</span>(tax_matrix)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>phylo_obj <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(ASV, TAX)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># phylo_obj</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Import metadata as sample data in phyloseq</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>samplenames <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(metadata_mcr)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># samplenames</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co"># join as phyloseq object</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>physeq_wnames <span class="ot">=</span> <span class="fu">merge_phyloseq</span>(phylo_obj, samplenames)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(ASV)</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co"># TAX</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Check</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>physeq_wnames</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 17878 taxa and 23 samples ]
sample_data() Sample Data:       [ 23 samples by 16 sample variables ]
tax_table()   Taxonomy Table:    [ 17878 taxa by 8 taxonomic ranks ]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ntaxa</span>(physeq_wnames) <span class="co">#17878</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 17878</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nsamples</span>(physeq_wnames) <span class="co">#23</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 23</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # physeq_wnames # run with sample, 1000 tax.</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # head(taxa_names(physeq_wnames))</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>mcr_tree <span class="ot">&lt;-</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(physeq_wnames), <span class="at">rooted =</span> <span class="cn">TRUE</span>, <span class="at">tip.label =</span> <span class="fu">taxa_names</span>(physeq_wnames))</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># class(mcr_tree)</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ?merge_phyloseq</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>physeq_mcr <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(physeq_wnames, mcr_tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="save-output-asv-tables-for-downstream-analyses" class="level3">
<h3 class="anchored" data-anchor-id="save-output-asv-tables-for-downstream-analyses">Save output ASV tables for downstream analyses</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># phylo_obj &lt;- phyloseq(ASV, TAX)</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># samplenames &lt;- sample_data(metadata_formatted)</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co"># physeq_wnames = merge_phyloseq(phylo_obj, samplenames)</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(asv_wtax_qc)</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># physeq_wnames</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># save(phylo_obj, samplenames, physeq_wnames, metadata_mcr, asv_wtax_qc, TAX, tax_matrix, physeq_mcr, file = "input-data/MCR-amplicon-data.RData")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sequence-stats" class="level2">
<h2 class="anchored" data-anchor-id="sequence-stats">Sequence stats</h2>
<p>Total ASVs and sequences for each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>asv_wtax_qc <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, VENT, SITE) <span class="sc">%&gt;%</span> </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM_stats =</span> <span class="fu">sum</span>(value),</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">ASV_stats =</span> <span class="fu">n_distinct</span>(FeatureID)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">ends_with</span>(<span class="st">"_stats"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_hline(yintercept=10000, linetype="dashed", color = "green") +</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> name, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE', 'VENT'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Create supplementary table with ASV and sequence stats.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>table_supp_seqstats <span class="ot">&lt;-</span> asv_wtax_qc <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, VENT, SITE) <span class="sc">%&gt;%</span> </span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM_stats =</span> <span class="fu">sum</span>(value),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">ASV_stats =</span> <span class="fu">n_distinct</span>(FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE', 'VENT'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(table_supp_seqstats, file = "output-data/supp-table-sequencestats.csv")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="visualize-metadata" class="level1">
<h1>Visualize metadata</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>env_params <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="at">file =</span> <span class="st">"table_wenv.txt"</span>, <span class="at">sep =</span> <span class="st">" "</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>subset_metadata <span class="ot">&lt;-</span> env_params <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">!=</span> <span class="st">"Incubation"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(VENT <span class="sc">!=</span> <span class="st">"Quakeplume"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SITE, VENT, SAMPLETYPE, RATE_min, EUK_ml, GRAZING_EFFECT_hr) <span class="sc">%&gt;%</span> </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE, VENT, SAMPLETYPE) <span class="sc">%&gt;%</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">RATE_min_avg =</span> <span class="fu">mean</span>(RATE_min),</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">EUK_ml_avg =</span> <span class="fu">mean</span>(EUK_ml),</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">GRAZING_EFFECT_hr_avg =</span> <span class="fu">mean</span>(GRAZING_EFFECT_hr)) <span class="sc">%&gt;%</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">ends_with</span>(<span class="st">"_avg"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SITE', 'VENT'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>metadata_numeric <span class="ot">&lt;-</span> metadata_mcr <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">!=</span> <span class="st">"Incubation"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(VENT, SITE, SAMPLETYPE, DEPTH, TEMP, pH, PercSeawater, Mg, H2, H2S, CH4, ProkConc) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> DEPTH<span class="sc">:</span>ProkConc, <span class="at">values_to =</span> <span class="st">"VALUE_TMP"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(VALUE_TMP)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>VALUE_TMP) <span class="sc">%&gt;%</span> </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(subset_metadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in mask$eval_all_mutate(quo): NAs introduced by coercion</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>tileplot_env <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  metadata_numeric <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">==</span> x) <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> name, <span class="at">y =</span> VENT, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(SITE <span class="sc">+</span> SAMPLETYPE <span class="sc">~</span> ., <span class="at">switch =</span> <span class="st">"both"</span>, <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scale =</span> <span class="st">"free"</span>) <span class="sc">+</span> <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">5</span>),</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_distiller</span>(<span class="at">palette =</span> <span class="st">"Reds"</span>, <span class="at">direction=</span><span class="dv">2</span>, <span class="at">na.value =</span> <span class="st">"grey50"</span>)</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_grid(</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("TEMP") + </span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "YlOrRd", direction=2, na.value = "grey50") +</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme(axis.text.y = element_text(color = "black", hjust = 1, vjust = 0), </span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co">#          strip.text.y = element_text(color = "black", hjust = 0, vjust = 0), </span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="co">#          strip.placement = "outside"),</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("PercSeawater") + </span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Blues", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("pH") + </span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "YlGnBu", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("Mg") + </span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Purples", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="co">#  # endemic_env("NO3"),</span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("H2") + </span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Purples", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("CH4") + </span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Purples", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("H2S") + </span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Purples", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("ProkConc") + </span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Oranges", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("EUK_ml_avg") + </span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a><span class="co">#    theme(axis.text.y = element_text(color = "black", hjust = 1, vjust = 0), </span></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a><span class="co">#          strip.text.y = element_text(color = "black", hjust = 0, vjust = 0), </span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a><span class="co">#          strip.placement = "outside") +</span></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Oranges", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a><span class="co">#  tileplot_env("GRAZING_EFFECT_hr_avg") + </span></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a><span class="co">#    scale_fill_distiller(palette = "Oranges", direction=2, na.value = "grey50"),</span></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a><span class="co">#  nrow = 1,</span></span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a><span class="co">#  rel_widths = c(4,1,1,1,1,1,1,1, 4, 1)</span></span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="community-structure-diversity-at-von-damm-piccard" class="level1">
<h1>Community structure &amp; diversity at Von Damm &amp; Piccard</h1>
<section id="cluster-analysis" class="level2">
<h2 class="anchored" data-anchor-id="cluster-analysis">Cluster analysis</h2>
<p>Ordination analysis</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan); <span class="fu">library</span>(ggdendro); <span class="fu">library</span>(compositions)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: permute</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>This is vegan 2.6-2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to compositions, a package for compositional data analysis.
Find an intro with "? compositions"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'compositions'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ape':

    balance</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    anova, cor, cov, dist, var</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    %*%, norm, scale, scale.default</code></pre>
</div>
</div>
</section>
<section id="cluster-analysis-with-grazing-assays" class="level2">
<h2 class="anchored" data-anchor-id="cluster-analysis-with-grazing-assays">Cluster analysis with grazing assays</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(asv_wtax_qc)</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>asv_mcr_numeric <span class="ot">&lt;-</span> asv_wtax_qc <span class="sc">%&gt;%</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(FeatureID, SAMPLENAME) <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MEAN_ACROSS_REPS =</span> <span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, SAMPLENAME, MEAN_ACROSS_REPS) <span class="sc">%&gt;%</span> </span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SAMPLENAME, <span class="at">values_from =</span> MEAN_ACROSS_REPS, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'FeatureID'. You can override using the
`.groups` argument.</code></pre>
</div>
</div>
<p>Transform compositional data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>logratio_mcr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(compositions<span class="sc">::</span><span class="fu">clr</span>(<span class="fu">t</span>(asv_mcr_numeric)))</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(logratio_mcr)</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co"># ?alr()</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ?ilr()</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>pca_logratio <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(logratio_mcr)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>variance_logratio <span class="ot">&lt;-</span> (pca_logratio<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)<span class="sc">/</span><span class="fu">sum</span>(pca_logratio<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(variance_logratio, <span class="at">main =</span> <span class="st">"Log-Ratio PCA Screeplot"</span>, <span class="at">xlab =</span> <span class="st">"PC Axis"</span>, <span class="at">ylab =</span> <span class="st">"% Variance"</span>, </span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.names =</span> <span class="fl">1.5</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>, <span class="at">cex.main =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract PCA points</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>mcr_pca_pts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pca_logratio<span class="sc">$</span>x, <span class="at">SAMPLE =</span> <span class="fu">rownames</span>(pca_logratio<span class="sc">$</span>x)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"SAMPLENAME"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(SAMPLENAME, <span class="fu">c</span>(<span class="st">"SITE"</span>, <span class="st">"SAMPLETYPE"</span>, <span class="st">"YEAR"</span>, <span class="st">"VENT"</span>), <span class="st">" "</span>, </span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">remove =</span> <span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 4 pieces. Additional pieces discarded in 21 rows [1, 2, 3, 4,
5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
</div>
</div>
<section id="pcoa-with-all-samples" class="level3">
<h3 class="anchored" data-anchor-id="pcoa-with-all-samples">PCoA with all samples</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>mcr_pca_pts <span class="sc">%&gt;%</span> </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TYPE =</span> <span class="fu">case_when</span>(</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>    SAMPLETYPE <span class="sc">==</span> <span class="st">"Incubation"</span> <span class="sc">~</span> <span class="st">"Incubation"</span>,</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"in situ"</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2)) <span class="sc">+</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> SITE, <span class="at">fill =</span> VENT, <span class="at">alpha =</span> TYPE)) <span class="sc">+</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">24</span>)) <span class="sc">+</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.4</span>)) <span class="sc">+</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="fu">c</span>(<span class="dv">22</span>)))) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Dendrogram analysis</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ?decostand()</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Relative abundance</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>rel_abun <span class="ot">&lt;-</span> <span class="fu">decostand</span>(asv_mcr_numeric, <span class="at">MARGIN =</span> <span class="dv">2</span>, <span class="at">method =</span> <span class="st">"total"</span>)</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Cluster dendrogram (average hierarchical clustering)</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>cluster_mcr <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(<span class="fu">t</span>(rel_abun)), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>dendro <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(cluster_mcr)</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>mcr_dendro <span class="ot">&lt;-</span> <span class="fu">dendro_data</span>(dendro, <span class="at">type =</span> <span class="st">"rectangle"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>mcr_dendro_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">segment</span>(mcr_dendro)) <span class="sc">+</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">xend =</span> xend, </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> yend)) <span class="sc">+</span> </span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_reverse</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.5</span>), <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>, <span class="fl">0.4</span>, <span class="fl">0.6</span>, <span class="fl">0.8</span>)) <span class="sc">+</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> label, <span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="dv">0</span>), <span class="at">data =</span> <span class="fu">label</span>(mcr_dendro)) <span class="sc">+</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_dendro</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Dissimilarity"</span>) <span class="sc">+</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">14</span>), <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"#252525"</span>), </span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">14</span>))</span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a><span class="co"># svg('figs/SUPPLEMENTARY-dendrogram-wreps.svg', w = 10, h = 8)</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>mcr_dendro_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="pcoa-with-in-situ-only" class="level3">
<h3 class="anchored" data-anchor-id="pcoa-with-in-situ-only">PCoA with in situ only</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>asv_mcr_numeric_insitu <span class="ot">&lt;-</span> asv_wtax_qc <span class="sc">%&gt;%</span> </span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">!=</span> <span class="st">"Incubation"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(FeatureID, SAMPLENAME) <span class="sc">%&gt;%</span> </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MEAN_ACROSS_REPS =</span> <span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, SAMPLENAME, MEAN_ACROSS_REPS) <span class="sc">%&gt;%</span> </span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SAMPLENAME, <span class="at">values_from =</span> MEAN_ACROSS_REPS, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'FeatureID'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>insitu_logratio_mcr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(compositions<span class="sc">::</span><span class="fu">clr</span>(<span class="fu">t</span>(asv_mcr_numeric_insitu)))</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>insitu_pca_logratio <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(insitu_logratio_mcr)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>insitu_variance_logratio <span class="ot">&lt;-</span> (insitu_pca_logratio<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)<span class="sc">/</span><span class="fu">sum</span>(insitu_pca_logratio<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(insitu_variance_logratio, <span class="at">main =</span> <span class="st">"Log-Ratio PCA Screeplot"</span>, <span class="at">xlab =</span> <span class="st">"PC Axis"</span>, <span class="at">ylab =</span> <span class="st">"% Variance"</span>, </span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.names =</span> <span class="fl">1.5</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>, <span class="at">cex.main =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract PCA points for only insitu samples</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>insitu_mcr_pca_pts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(insitu_pca_logratio<span class="sc">$</span>x, <span class="at">SAMPLE =</span> <span class="fu">rownames</span>(insitu_pca_logratio<span class="sc">$</span>x)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"SAMPLENAME"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(SAMPLENAME, <span class="fu">c</span>(<span class="st">"SITE"</span>, <span class="st">"SAMPLETYPE"</span>, <span class="st">"YEAR"</span>, <span class="st">"VENT"</span>), <span class="st">" "</span>, </span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">remove =</span> <span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 4 pieces. Additional pieces discarded in 13 rows [1, 2, 3, 4,
5, 6, 7, 8, 9, 10, 11, 12, 13].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>insitu_mcr_pca_pts <span class="sc">%&gt;%</span> </span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TYPE =</span> <span class="fu">case_when</span>(</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>    SAMPLETYPE <span class="sc">==</span> <span class="st">"Incubation"</span> <span class="sc">~</span> <span class="st">"Incubation"</span>,</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"in situ"</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2)) <span class="sc">+</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> SITE, <span class="at">fill =</span> VENT, <span class="at">alpha =</span> TYPE)) <span class="sc">+</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">24</span>)) <span class="sc">+</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.4</span>)) <span class="sc">+</span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="fu">c</span>(<span class="dv">22</span>)))) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="768"></p>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep same sample colors to compare to cluster analysis with in situ and Tf samples.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="shared-asvs" class="level2">
<h2 class="anchored" data-anchor-id="shared-asvs">Shared ASVs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/MCR-amplicon-data.RData"</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  phylo_obj
  samplenames
  physeq_wnames
  metadata_mcr
  asv_wtax_qc
  TAX
  tax_matrix
  physeq_mcr</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(asv_wtax_qc$SITE)</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(asv_wtax_qc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  21
  FeatureID SAMPLE value Taxon Conse SAMPL VENT  SITE  SAMPL DEPTH SAMPL
  &lt;chr&gt;     &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  
1 00056209 52_MC     0 Euka     0.8 "Picca Lots Picc ""      ""    Incuba
2 00056209 53_MC     0 Euka     0.8 "Picca Lots Picc ""      ""    Incuba
3 00056209 54_MC     0 Euka     0.8 "VonDa Must VonD ""      ""    Incuba
4 00056209 55_MC     0 Euka     0.8 "VonDa Shri VonD ""      ""    Incuba
5 00056209 57_MC     0 Euka     0.8 "Picca Shri Picc ""      ""    Incuba
6 00056209 58_MC     0 Euka     0.8 "VonDa Plume VonD ""      ""    Incuba
#  with 10 more variables: YEAR &lt;chr&gt;, TEMP &lt;chr&gt;, pH &lt;chr&gt;,
#   PercSeawater &lt;chr&gt;, Mg &lt;chr&gt;, H2 &lt;chr&gt;, H2S &lt;chr&gt;, CH4 &lt;chr&gt;,
#   ProkConc &lt;chr&gt;, Sample_or_Control &lt;chr&gt;, and abbreviated variable names
#   Consensus, SAMPLENAME, SAMPLEID, SAMPLETYPE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(TAX)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Function to generate upsetR plot with varied taxonomic levels.</p>
<p>Questions:</p>
<ul>
<li><p>How many ASVs are shared among vent sites within Von Damm and Piccard?</p></li>
<li><p>What about shared ASVs across each vent field?</p></li>
<li><p>Is there a taxonomic level that drives site-to-site or field-to-field community dissimilarity?</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | fig-width: 10</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | fig-height: 15</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="co"># options for taxa: SupergroupPhylum, Supergroup, Phylum, Class, Order, Family, Genus, Species</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>alv <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alveolata-Ellobiopsidae"</span>, <span class="st">"Alveolata-Perkinsea"</span>, <span class="st">"Alveolata-Unknown"</span>, <span class="st">"Alveolata-Chrompodellids"</span>, <span class="st">"Alveolata-Apicomplexa"</span>)</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>make_upset_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(df, taxa_level){ </span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>  taxa_select <span class="ot">&lt;-</span> <span class="fu">enquo</span>(taxa_level)</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> </span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">!=</span> <span class="st">"Incubation"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Taxon, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Supergroup"</span>, </span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,</span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> <span class="co">#select eukaryotes only</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Supergroup <span class="sc">!=</span> <span class="st">"Opisthokonta"</span>) <span class="sc">%&gt;%</span> <span class="co"># remove multicellular metazoa</span></span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Supergroup =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Supergroup), <span class="st">"Unknown Eukaryota"</span>, Supergroup),</span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">Phylum =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Phylum), <span class="st">"Unknown"</span>, Phylum),</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">Phylum =</span> <span class="fu">ifelse</span>(Phylum <span class="sc">==</span> <span class="st">"Alveolata_X"</span>, <span class="st">"Ellobiopsidae"</span>, Phylum),</span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">Supergroup =</span> <span class="fu">ifelse</span>(Supergroup <span class="sc">==</span> <span class="st">"Alveolata"</span>, <span class="fu">paste</span>(Supergroup, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>), Supergroup)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">case_when</span>(</span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">%in%</span> alv <span class="sc">~</span> <span class="st">"Other Alveolata"</span>,</span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Eukaryota_X"</span> <span class="sc">~</span> <span class="st">"Unknown Eukaryota"</span>,</span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Cercozoa"</span> <span class="sc">~</span> <span class="st">"Rhizaria-Cercozoa"</span>,</span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Radiolaria"</span> <span class="sc">~</span> <span class="st">"Rhizaria-Radiolaria"</span>,</span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Ochrophyta"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Ochrophyta"</span>,</span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Opalozoa"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Opalozoa"</span>,</span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Sagenista"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Sagenista"</span>,</span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> Supergroup</span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb56-30"><a href="#cb56-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Taxa to supergroup</span></span>
<span id="cb56-31"><a href="#cb56-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SupergroupPhylum =</span> SUPERGROUP) <span class="sc">%&gt;%</span> <span class="co">#add modified "supergroup-phylum category"</span></span>
<span id="cb56-32"><a href="#cb56-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Average across replicates</span></span>
<span id="cb56-33"><a href="#cb56-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(FeatureID, SAMPLENAME, VENT, <span class="sc">!!</span>taxa_select) <span class="sc">%&gt;%</span> </span>
<span id="cb56-34"><a href="#cb56-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">AVG =</span> <span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-35"><a href="#cb56-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb56-36"><a href="#cb56-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(SAMPLENAME, <span class="fu">c</span>(<span class="st">"SITE"</span>, <span class="st">"SAMPLETYPE"</span>, <span class="st">"YEAR"</span>, <span class="st">"Sample_tmp"</span>), <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-37"><a href="#cb56-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REGION =</span> <span class="st">"Mid-Cayman Rise"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb56-38"><a href="#cb56-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">VENTNAME =</span> <span class="fu">paste</span>(SITE, VENT, <span class="at">sep =</span> <span class="st">" "</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-39"><a href="#cb56-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>Sample_tmp) <span class="sc">%&gt;%</span> </span>
<span id="cb56-40"><a href="#cb56-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(SAMPLE, SITE, SAMPLETYPE, VENT, <span class="at">sep =</span> <span class="st">" "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-41"><a href="#cb56-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb56-42"><a href="#cb56-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="sc">!!</span>taxa_select, SAMPLE) <span class="sc">%&gt;%</span> </span>
<span id="cb56-43"><a href="#cb56-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(AVG)) <span class="sc">%&gt;%</span></span>
<span id="cb56-44"><a href="#cb56-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-45"><a href="#cb56-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(<span class="sc">!!</span>taxa_select, SUM, SAMPLE, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-46"><a href="#cb56-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="sc">!!</span>taxa_select) <span class="sc">%&gt;%</span> </span>
<span id="cb56-47"><a href="#cb56-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SAMPLE =</span> <span class="fu">list</span>(SAMPLE)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-48"><a href="#cb56-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE)) <span class="sc">+</span></span>
<span id="cb56-49"><a href="#cb56-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">width =</span> <span class="fl">0.7</span>, <span class="fu">aes</span>(<span class="at">fill =</span> <span class="sc">!!</span>taxa_select)) <span class="sc">+</span></span>
<span id="cb56-50"><a href="#cb56-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_upset</span>(<span class="at">n_intersections =</span> <span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb56-51"><a href="#cb56-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb56-52"><a href="#cb56-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Shared at taxonomic level"</span>) <span class="sc">+</span></span>
<span id="cb56-53"><a href="#cb56-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb56-54"><a href="#cb56-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">14</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb56-55"><a href="#cb56-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">14</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb56-56"><a href="#cb56-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">14</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb56-57"><a href="#cb56-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb56-58"><a href="#cb56-58" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb56-59"><a href="#cb56-59" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb56-60"><a href="#cb56-60" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">5</span>, <span class="st">"cm"</span>))</span>
<span id="cb56-61"><a href="#cb56-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># +</span></span>
<span id="cb56-62"><a href="#cb56-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_fill_manual(values = all_taxa_color)</span></span>
<span id="cb56-63"><a href="#cb56-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb56-64"><a href="#cb56-64" aria-hidden="true" tabindex="-1"></a><span class="co"># all_taxa_color</span></span>
<span id="cb56-65"><a href="#cb56-65" aria-hidden="true" tabindex="-1"></a><span class="co">#values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#deebf7", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")</span></span>
<span id="cb56-66"><a href="#cb56-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data to reduce noise and show sample type to vent ecosystem variability.</span></span>
<span id="cb56-67"><a href="#cb56-67" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Example comparisons of shared taxonomic levels. <em>Explanation</em>: At ASV level, most ASVs are not shared across samples. So at other taxonomic categories, are these taxa also present or are there whole classes not present?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | fig-width: 10</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | fig-height: 15</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="co"># SupergroupPhylum, Supergroup, Phylum, Class, Order, Family, Genus, Species</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="co"># make_upset_plot(asv_wtax_qc, SupergroupPhylum)</span></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a><span class="co"># make_upset_plot(asv_wtax_qc, Class) + theme(legend.position = "none")</span></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a><span class="co"># make_upset_plot(asv_wtax_qc, Supergroup) + theme(legend.position = "none")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(asv_wtax_qc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  21
  FeatureID SAMPLE value Taxon Conse SAMPL VENT  SITE  SAMPL DEPTH SAMPL
  &lt;chr&gt;     &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  
1 00056209 52_MC     0 Euka     0.8 "Picca Lots Picc ""      ""    Incuba
2 00056209 53_MC     0 Euka     0.8 "Picca Lots Picc ""      ""    Incuba
3 00056209 54_MC     0 Euka     0.8 "VonDa Must VonD ""      ""    Incuba
4 00056209 55_MC     0 Euka     0.8 "VonDa Shri VonD ""      ""    Incuba
5 00056209 57_MC     0 Euka     0.8 "Picca Shri Picc ""      ""    Incuba
6 00056209 58_MC     0 Euka     0.8 "VonDa Plume VonD ""      ""    Incuba
#  with 10 more variables: YEAR &lt;chr&gt;, TEMP &lt;chr&gt;, pH &lt;chr&gt;,
#   PercSeawater &lt;chr&gt;, Mg &lt;chr&gt;, H2 &lt;chr&gt;, H2S &lt;chr&gt;, CH4 &lt;chr&gt;,
#   ProkConc &lt;chr&gt;, Sample_or_Control &lt;chr&gt;, and abbreviated variable names
#   Consensus, SAMPLENAME, SAMPLEID, SAMPLETYPE</code></pre>
</div>
</div>
<p>A better visualization may be a presence/absence across samples with a tile plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>alv <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alveolata-Ellobiopsidae"</span>, <span class="st">"Alveolata-Perkinsea"</span>, <span class="st">"Alveolata-Unknown"</span>, <span class="st">"Alveolata-Chrompodellids"</span>, <span class="st">"Alveolata-Apicomplexa"</span>)</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>make_tile_plot_pa <span class="ot">&lt;-</span> <span class="cf">function</span>(df, taxa_level, level0){ </span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  taxa_select <span class="ot">&lt;-</span> <span class="fu">enquo</span>(taxa_level)</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>  taxa_select_higher <span class="ot">&lt;-</span> <span class="fu">enquo</span>(level0)</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> </span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">!=</span> <span class="st">"Incubation"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Taxon, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Supergroup"</span>, </span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>,</span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> <span class="co">#select eukaryotes only</span></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Supergroup <span class="sc">!=</span> <span class="st">"Opisthokonta"</span>) <span class="sc">%&gt;%</span> <span class="co"># remove multicellular metazoa</span></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Supergroup =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Supergroup), <span class="st">"Unknown Eukaryota"</span>, Supergroup),</span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">Phylum =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Phylum), <span class="st">"Unknown"</span>, Phylum),</span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">Phylum =</span> <span class="fu">ifelse</span>(Phylum <span class="sc">==</span> <span class="st">"Alveolata_X"</span>, <span class="st">"Ellobiopsidae"</span>, Phylum),</span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">Supergroup =</span> <span class="fu">ifelse</span>(Supergroup <span class="sc">==</span> <span class="st">"Alveolata"</span>, <span class="fu">paste</span>(Supergroup, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>), Supergroup)) <span class="sc">%&gt;%</span> </span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">case_when</span>(</span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">%in%</span> alv <span class="sc">~</span> <span class="st">"Other Alveolata"</span>,</span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Eukaryota_X"</span> <span class="sc">~</span> <span class="st">"Unknown Eukaryota"</span>,</span>
<span id="cb60-21"><a href="#cb60-21" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Cercozoa"</span> <span class="sc">~</span> <span class="st">"Rhizaria-Cercozoa"</span>,</span>
<span id="cb60-22"><a href="#cb60-22" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Radiolaria"</span> <span class="sc">~</span> <span class="st">"Rhizaria-Radiolaria"</span>,</span>
<span id="cb60-23"><a href="#cb60-23" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Ochrophyta"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Ochrophyta"</span>,</span>
<span id="cb60-24"><a href="#cb60-24" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Opalozoa"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Opalozoa"</span>,</span>
<span id="cb60-25"><a href="#cb60-25" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Sagenista"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Sagenista"</span>,</span>
<span id="cb60-26"><a href="#cb60-26" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> Supergroup</span>
<span id="cb60-27"><a href="#cb60-27" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb60-28"><a href="#cb60-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Taxa to supergroup</span></span>
<span id="cb60-29"><a href="#cb60-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SupergroupPhylum =</span> SUPERGROUP) <span class="sc">%&gt;%</span> <span class="co">#add modified "supergroup-phylum category"</span></span>
<span id="cb60-30"><a href="#cb60-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Average across replicates</span></span>
<span id="cb60-31"><a href="#cb60-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(FeatureID, SAMPLENAME, VENT, <span class="sc">!!</span>taxa_select_higher, <span class="sc">!!</span>taxa_select) <span class="sc">%&gt;%</span> </span>
<span id="cb60-32"><a href="#cb60-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">AVG =</span> <span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb60-33"><a href="#cb60-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb60-34"><a href="#cb60-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(SAMPLENAME, <span class="fu">c</span>(<span class="st">"SITE"</span>, <span class="st">"SAMPLETYPE"</span>, <span class="st">"YEAR"</span>, <span class="st">"Sample_tmp"</span>), <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-35"><a href="#cb60-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REGION =</span> <span class="st">"Mid-Cayman Rise"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb60-36"><a href="#cb60-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">VENTNAME =</span> <span class="fu">paste</span>(SITE, VENT, <span class="at">sep =</span> <span class="st">" "</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb60-37"><a href="#cb60-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>Sample_tmp) <span class="sc">%&gt;%</span> </span>
<span id="cb60-38"><a href="#cb60-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(SAMPLE, SITE, SAMPLETYPE, VENT, <span class="at">sep =</span> <span class="st">" "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-39"><a href="#cb60-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE, <span class="sc">!!</span>taxa_select_higher, <span class="sc">!!</span>taxa_select, SAMPLE) <span class="sc">%&gt;%</span> </span>
<span id="cb60-40"><a href="#cb60-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(AVG)) <span class="sc">%&gt;%</span></span>
<span id="cb60-41"><a href="#cb60-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb60-42"><a href="#cb60-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_column</span>(<span class="at">PRESENT =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-43"><a href="#cb60-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> <span class="co"># remove NAs in taxa categories</span></span>
<span id="cb60-44"><a href="#cb60-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> <span class="sc">!!</span>taxa_select)) <span class="sc">+</span></span>
<span id="cb60-45"><a href="#cb60-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="st">"darkgrey"</span>, <span class="at">width =</span> <span class="fl">0.6</span>, <span class="fu">aes</span>(<span class="at">fill =</span> PRESENT)) <span class="sc">+</span></span>
<span id="cb60-46"><a href="#cb60-46" aria-hidden="true" tabindex="-1"></a>      <span class="fu">facet_grid</span>(SupergroupPhylum <span class="sc">~</span> SITE, <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb60-47"><a href="#cb60-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb60-48"><a href="#cb60-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">12</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb60-49"><a href="#cb60-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">12</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb60-50"><a href="#cb60-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">12</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb60-51"><a href="#cb60-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb60-52"><a href="#cb60-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb60-53"><a href="#cb60-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb60-54"><a href="#cb60-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">5</span>, <span class="st">"cm"</span>))</span>
<span id="cb60-55"><a href="#cb60-55" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | fig-width: 10</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | fig-height: 20</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="fu">make_tile_plot_pa</span>(asv_wtax_qc, Class, SupergroupPhylum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 8 pieces. Additional pieces discarded in 5926 rows [5, 6, 7,
8, 9, 10, 11, 14, 15, 16, 17, 20, 21, 22, 24, 25, 26, 28, 29, 30, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 8 pieces. Missing pieces filled with `NA` in 3451 rows [1, 2,
3, 4, 12, 13, 18, 19, 23, 27, 35, 44, 45, 46, 48, 49, 53, 54, 55, 56, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'FeatureID', 'SAMPLENAME', 'VENT',
'SupergroupPhylum'. You can override using the `.groups` argument.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 4 pieces. Additional pieces discarded in 7678 rows [1, 2, 3,
4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SITE', 'SupergroupPhylum', 'Class'. You
can override using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="richness-by-site" class="level2">
<h2 class="anchored" data-anchor-id="richness-by-site">Richness by site</h2>
</section>
<section id="corncob" class="level2">
<h2 class="anchored" data-anchor-id="corncob">Corncob</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"bryandmartin/corncob"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Skipping install of 'corncob' from a github remote, the SHA1 (950f4f0a) has not changed since last install.
  Use `force = TRUE` to force installation</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corncob); <span class="fu">library</span>(phyloseq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/MCR-amplicon-data.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  phylo_obj
  samplenames
  physeq_wnames
  metadata_mcr
  asv_wtax_qc
  TAX
  tax_matrix
  physeq_mcr</code></pre>
</div>
</div>
<p>Explore data input for corncob</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">otu_table</span>(physeq_mcr)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>OTU Table:          [3 taxa and 3 samples]
                     taxa are rows
                                 52_MCR_Piccard_expTf_LotsOShrimp_A_0_Jun2021
9d4a549902de604c74ab0c60bb2b1b90                                            0
8d216a4d59a36e484c3cc570f2d6091c                                            0
86eb0e2a76681d0a5aada91615bfc041                                            0
                                 53_MCR_Piccard_expTf_LotsOShrimp_B_0_Jun2021
9d4a549902de604c74ab0c60bb2b1b90                                            0
8d216a4d59a36e484c3cc570f2d6091c                                            0
86eb0e2a76681d0a5aada91615bfc041                                            0
                                 54_MCR_VonDamm_expTf_MustardStand_0_0_Jun2021
9d4a549902de604c74ab0c60bb2b1b90                                             0
8d216a4d59a36e484c3cc570f2d6091c                                             0
86eb0e2a76681d0a5aada91615bfc041                                             0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(physeq_mcr)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                              X         VENT    SITE SAMPLEID
52_MCR_Piccard_expTf_LotsOShrimp_A_0_Jun2021  2  LotsOShrimp Piccard         
53_MCR_Piccard_expTf_LotsOShrimp_B_0_Jun2021  3  LotsOShrimp Piccard         
54_MCR_VonDamm_expTf_MustardStand_0_0_Jun2021 9 MustardStand VonDamm         
                                              DEPTH SAMPLETYPE YEAR TEMP pH
52_MCR_Piccard_expTf_LotsOShrimp_A_0_Jun2021        Incubation 2020        
53_MCR_Piccard_expTf_LotsOShrimp_B_0_Jun2021        Incubation 2020        
54_MCR_VonDamm_expTf_MustardStand_0_0_Jun2021       Incubation 2020        
                                              PercSeawater Mg H2 H2S CH4
52_MCR_Piccard_expTf_LotsOShrimp_A_0_Jun2021          &lt;NA&gt;  0           
53_MCR_Piccard_expTf_LotsOShrimp_B_0_Jun2021          &lt;NA&gt;  0           
54_MCR_VonDamm_expTf_MustardStand_0_0_Jun2021         &lt;NA&gt;  0           
                                              ProkConc Sample_or_Control
52_MCR_Piccard_expTf_LotsOShrimp_A_0_Jun2021                      Sample
53_MCR_Piccard_expTf_LotsOShrimp_B_0_Jun2021                      Sample
54_MCR_VonDamm_expTf_MustardStand_0_0_Jun2021                     Sample</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(physeq_mcr)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Taxonomy Table:     [3 taxa by 8 taxonomic ranks]:
                                 Domain       Supergroup  Phylum      
9d4a549902de604c74ab0c60bb2b1b90 "Unassigned" NA          NA          
8d216a4d59a36e484c3cc570f2d6091c "Eukaryota"  "Alveolata" "Ciliophora"
86eb0e2a76681d0a5aada91615bfc041 "Eukaryota"  "Alveolata" "Ciliophora"
                                 Class          Order          Family        
9d4a549902de604c74ab0c60bb2b1b90 NA             NA             NA            
8d216a4d59a36e484c3cc570f2d6091c "Spirotrichea" "Euplotia"     "Aspidiscidae"
86eb0e2a76681d0a5aada91615bfc041 "Armophorea"   "Armophorea_X" "Metopidae"   
                                 Genus       Species      
9d4a549902de604c74ab0c60bb2b1b90 NA          NA           
8d216a4d59a36e484c3cc570f2d6091c "Aspidisca" NA           
86eb0e2a76681d0a5aada91615bfc041 "Metopus_1" "Metopus_sp."</code></pre>
</div>
</div>
</section>
</section>
<section id="compare-in-situ-and-tf-samples" class="level1">
<h1>Compare in situ and Tf samples</h1>
<p>Examine in situ to T final samples for presence/absence.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(asv_wtax_qc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  21
  FeatureID SAMPLE value Taxon Conse SAMPL VENT  SITE  SAMPL DEPTH SAMPL
  &lt;chr&gt;     &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  
1 00056209 52_MC     0 Euka     0.8 "Picca Lots Picc ""      ""    Incuba
2 00056209 53_MC     0 Euka     0.8 "Picca Lots Picc ""      ""    Incuba
3 00056209 54_MC     0 Euka     0.8 "VonDa Must VonD ""      ""    Incuba
4 00056209 55_MC     0 Euka     0.8 "VonDa Shri VonD ""      ""    Incuba
5 00056209 57_MC     0 Euka     0.8 "Picca Shri Picc ""      ""    Incuba
6 00056209 58_MC     0 Euka     0.8 "VonDa Plume VonD ""      ""    Incuba
#  with 10 more variables: YEAR &lt;chr&gt;, TEMP &lt;chr&gt;, pH &lt;chr&gt;,
#   PercSeawater &lt;chr&gt;, Mg &lt;chr&gt;, H2 &lt;chr&gt;, H2S &lt;chr&gt;, CH4 &lt;chr&gt;,
#   ProkConc &lt;chr&gt;, Sample_or_Control &lt;chr&gt;, and abbreviated variable names
#   Consensus, SAMPLENAME, SAMPLEID, SAMPLETYPE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># View(asv_wtax_qc)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Calculate percent change from in situ to Tf for all grazing experiments</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>alv <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alveolata-Ellobiopsidae"</span>, <span class="st">"Alveolata-Perkinsea"</span>, <span class="st">"Alveolata-Unknown"</span>, <span class="st">"Alveolata-Chrompodellids"</span>, <span class="st">"Alveolata-Apicomplexa"</span>)</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract taxa information</span></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>taxa_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(TAX)) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>)</span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>perc_change <span class="ot">&lt;-</span> asv_wtax_qc <span class="sc">%&gt;%</span> </span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Avg seq count by ASV across replicates</span></span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE, VENT, SAMPLETYPE, FeatureID) <span class="sc">%&gt;%</span> </span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">avg_seq =</span> <span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb81-11"><a href="#cb81-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ASSAY =</span> <span class="fu">case_when</span>(</span>
<span id="cb81-12"><a href="#cb81-12" aria-hidden="true" tabindex="-1"></a>    SAMPLETYPE <span class="sc">==</span> <span class="st">"Incubation"</span> <span class="sc">~</span> <span class="st">"Tf"</span>,</span>
<span id="cb81-13"><a href="#cb81-13" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"insitu"</span></span>
<span id="cb81-14"><a href="#cb81-14" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb81-15"><a href="#cb81-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb81-16"><a href="#cb81-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>SAMPLETYPE) <span class="sc">%&gt;%</span></span>
<span id="cb81-17"><a href="#cb81-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> ASSAY, <span class="at">values_from =</span> avg_seq, <span class="at">values_fn =</span> sum) <span class="sc">%&gt;%</span> </span>
<span id="cb81-18"><a href="#cb81-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select only ASVs present in both in situ and incubation experiments.</span></span>
<span id="cb81-19"><a href="#cb81-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> <span class="co"># remove NAs</span></span>
<span id="cb81-20"><a href="#cb81-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(insitu <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> Tf <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> <span class="co"># remove if 1 or the other are zero</span></span>
<span id="cb81-21"><a href="#cb81-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(insitu <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> Tf <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb81-22"><a href="#cb81-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">seq_change =</span> Tf <span class="sc">-</span> insitu) <span class="sc">%&gt;%</span> </span>
<span id="cb81-23"><a href="#cb81-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">category_change =</span> <span class="fu">case_when</span>(</span>
<span id="cb81-24"><a href="#cb81-24" aria-hidden="true" tabindex="-1"></a>           seq_change <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"Decrease"</span>,</span>
<span id="cb81-25"><a href="#cb81-25" aria-hidden="true" tabindex="-1"></a>           seq_change <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"Increase"</span>,</span>
<span id="cb81-26"><a href="#cb81-26" aria-hidden="true" tabindex="-1"></a>           seq_change <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"No Change"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb81-27"><a href="#cb81-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add in supergroup information</span></span>
<span id="cb81-28"><a href="#cb81-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">select</span>(taxa_table, FeatureID, Domain<span class="sc">:</span>Species) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb81-29"><a href="#cb81-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb81-30"><a href="#cb81-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Supergroup <span class="sc">!=</span> <span class="st">"Opisthokonta"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb81-31"><a href="#cb81-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Supergroup =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Supergroup), <span class="st">"Unknown Eukaryota"</span>, Supergroup),</span>
<span id="cb81-32"><a href="#cb81-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">Phylum =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Phylum), <span class="st">"Unknown"</span>, Phylum),</span>
<span id="cb81-33"><a href="#cb81-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">Phylum =</span> <span class="fu">ifelse</span>(Phylum <span class="sc">==</span> <span class="st">"Alveolata_X"</span>, <span class="st">"Ellobiopsidae"</span>, Phylum),</span>
<span id="cb81-34"><a href="#cb81-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">Supergroup =</span> <span class="fu">ifelse</span>(Supergroup <span class="sc">==</span> <span class="st">"Alveolata"</span>, <span class="fu">paste</span>(Supergroup, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>), Supergroup)) <span class="sc">%&gt;%</span> </span>
<span id="cb81-35"><a href="#cb81-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">case_when</span>(</span>
<span id="cb81-36"><a href="#cb81-36" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">%in%</span> alv <span class="sc">~</span> <span class="st">"Other Alveolata"</span>,</span>
<span id="cb81-37"><a href="#cb81-37" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Eukaryota_X"</span> <span class="sc">~</span> <span class="st">"Unknown Eukaryota"</span>,</span>
<span id="cb81-38"><a href="#cb81-38" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Cercozoa"</span> <span class="sc">~</span> <span class="st">"Rhizaria-Cercozoa"</span>,</span>
<span id="cb81-39"><a href="#cb81-39" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Radiolaria"</span> <span class="sc">~</span> <span class="st">"Rhizaria-Radiolaria"</span>,</span>
<span id="cb81-40"><a href="#cb81-40" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Ochrophyta"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Ochrophyta"</span>,</span>
<span id="cb81-41"><a href="#cb81-41" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Opalozoa"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Opalozoa"</span>,</span>
<span id="cb81-42"><a href="#cb81-42" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Sagenista"</span> <span class="sc">~</span> <span class="st">"Stramenopiles-Sagenista"</span>,</span>
<span id="cb81-43"><a href="#cb81-43" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> Supergroup</span>
<span id="cb81-44"><a href="#cb81-44" aria-hidden="true" tabindex="-1"></a>  ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SITE', 'VENT', 'SAMPLETYPE'. You can
override using the `.groups` argument.
Joining, by = "FeatureID"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(perc_change<span class="sc">$</span>category_change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 Decrease  Increase No Change 
     1073      1291        28 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(perc_change)</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(perc_change[1:2])</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Of the ASVs that appeared in both in situ and Tf for each grazing experiment, 1073 decreased in sequence relative abundance, and over 1200 increased. Only 28 showed no change in sequence relative abundance.</p>
<p>Formatted table to show sequence increase vs.&nbsp;decrease PER ASV from insitu to experiment Tf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>all_taxa_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alveolata-Ciliophora"</span>, <span class="st">"Alveolata-Dinoflagellata"</span>, <span class="st">"Protalveolata"</span>, <span class="st">"Other Alveolata"</span>, <span class="st">"Amoebozoa"</span>, <span class="st">"Apusozoa"</span>, <span class="st">"Excavata"</span>, <span class="st">"Hacrobia"</span>, <span class="st">"Archaeplastida"</span>, <span class="st">"Rhizaria"</span>, <span class="st">"Rhizaria-Radiolaria"</span>, <span class="st">"Rhizaria-Cercozoa"</span>, <span class="st">"Stramenopiles"</span>, <span class="st">"Stramenopiles-Opalozoa"</span>, <span class="st">"Stramenopiles-Sagenista"</span>, <span class="st">"Stramenopiles-Ochrophyta"</span>, <span class="st">"Opisthokonta"</span>, <span class="st">"Unknown Eukaryota"</span>)</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="co"># length(all_taxa_order)</span></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>all_taxa_color <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#fa9fb5"</span>, <span class="st">"#c51b8a"</span>, <span class="st">"#67000d"</span>, <span class="st">"#ef3b2c"</span>, <span class="st">"#ffffcc"</span>, <span class="st">"#feb24c"</span>,<span class="st">"#cc4c02"</span>, <span class="st">"#c7e9b4"</span>, <span class="st">"#238b45"</span>, <span class="st">"#c6dbef"</span>, <span class="st">"#99d8c9"</span>,<span class="st">"#1d91c0"</span>, <span class="st">"#253494"</span>, <span class="st">"#9e9ac8"</span>, <span class="st">"#54278f"</span>, <span class="st">"#bdbdbd"</span>, <span class="st">"#252525"</span>, <span class="st">"#f0f0f0"</span>)</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a><span class="co"># length(all_taxa_color)</span></span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>perc_change <span class="sc">%&gt;%</span> </span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE, VENT, category_change, SUPERGROUP) <span class="sc">%&gt;%</span> </span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sum_change =</span> <span class="fu">sum</span>(seq_change),</span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">asv_total =</span> <span class="fu">n_distinct</span>(FeatureID)) <span class="sc">%&gt;%</span> </span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP_ORDER =</span> <span class="fu">factor</span>(SUPERGROUP, <span class="at">levels =</span> all_taxa_order)) <span class="sc">%&gt;%</span> </span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> sum_change, <span class="at">y =</span> VENT, <span class="at">fill =</span> SUPERGROUP_ORDER)) <span class="sc">+</span></span>
<span id="cb86-13"><a href="#cb86-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">width =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb86-14"><a href="#cb86-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">150000</span>, <span class="dv">150000</span>)) <span class="sc">+</span></span>
<span id="cb86-15"><a href="#cb86-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> all_taxa_color) <span class="sc">+</span></span>
<span id="cb86-16"><a href="#cb86-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(SITE <span class="sc">~</span> ., <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb86-17"><a href="#cb86-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb86-18"><a href="#cb86-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb86-19"><a href="#cb86-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb86-20"><a href="#cb86-20" aria-hidden="true" tabindex="-1"></a>          <span class="co"># strip.text = element_blank(),</span></span>
<span id="cb86-21"><a href="#cb86-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb86-22"><a href="#cb86-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb86-23"><a href="#cb86-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb86-24"><a href="#cb86-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb86-25"><a href="#cb86-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb86-26"><a href="#cb86-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">5</span>),</span>
<span id="cb86-27"><a href="#cb86-27" aria-hidden="true" tabindex="-1"></a>          <span class="co"># panel.background = element_blank(),</span></span>
<span id="cb86-28"><a href="#cb86-28" aria-hidden="true" tabindex="-1"></a>          <span class="co"># panel.border = element_blank(),</span></span>
<span id="cb86-29"><a href="#cb86-29" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb86-30"><a href="#cb86-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sequence Decrease           Sequence Increase"</span>, <span class="at">y =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SITE', 'VENT', 'category_change'. You can
override using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="864"></p>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_bar(filter(category_change == "Increase"), stat = "identity") </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># str(perc_change)</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>perc_change <span class="sc">%&gt;%</span> </span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE, VENT, category_change, SUPERGROUP) <span class="sc">%&gt;%</span> </span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sum_change =</span> <span class="fu">sum</span>(seq_change),</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">asv_total =</span> <span class="fu">n_distinct</span>(FeatureID)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">asv_int =</span> <span class="fu">as.numeric</span>(asv_total)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">asv_change =</span> <span class="fu">case_when</span>(</span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>           category_change <span class="sc">==</span> <span class="st">"Decrease"</span> <span class="sc">~</span> (<span class="dv">0</span> <span class="sc">-</span> asv_int),</span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>           <span class="cn">TRUE</span> <span class="sc">~</span> asv_int</span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a>         )) <span class="sc">%&gt;%</span> </span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP_ORDER =</span> <span class="fu">factor</span>(SUPERGROUP, <span class="at">levels =</span> all_taxa_order)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-13"><a href="#cb89-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> asv_change, <span class="at">y =</span> VENT, <span class="at">fill =</span> SUPERGROUP_ORDER)) <span class="sc">+</span></span>
<span id="cb89-14"><a href="#cb89-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb89-15"><a href="#cb89-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">350</span>, <span class="dv">350</span>)) <span class="sc">+</span></span>
<span id="cb89-16"><a href="#cb89-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> all_taxa_color) <span class="sc">+</span></span>
<span id="cb89-17"><a href="#cb89-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(SITE <span class="sc">~</span> ., <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb89-18"><a href="#cb89-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb89-19"><a href="#cb89-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb89-20"><a href="#cb89-20" aria-hidden="true" tabindex="-1"></a>          <span class="co"># strip.text = element_blank(),</span></span>
<span id="cb89-21"><a href="#cb89-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-22"><a href="#cb89-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-23"><a href="#cb89-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb89-24"><a href="#cb89-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-25"><a href="#cb89-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb89-26"><a href="#cb89-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">5</span>),</span>
<span id="cb89-27"><a href="#cb89-27" aria-hidden="true" tabindex="-1"></a>          <span class="co"># panel.background = element_blank(),</span></span>
<span id="cb89-28"><a href="#cb89-28" aria-hidden="true" tabindex="-1"></a>          <span class="co"># panel.border = element_blank(),</span></span>
<span id="cb89-29"><a href="#cb89-29" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb89-30"><a href="#cb89-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"ASV change"</span>, <span class="at">y =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SITE', 'VENT', 'category_change'. You can
override using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="asv_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="session-information" class="level1">
<h1>Session information</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] corncob_0.3.0      compositions_2.0-4 ggdendro_0.1.23    vegan_2.6-2       
 [5] lattice_0.20-45    permute_0.9-7      ape_5.6-2          phyloseq_1.40.0   
 [9] forcats_0.5.2      stringr_1.4.1      dplyr_1.0.10       purrr_0.3.5       
[13] readr_2.1.2        tidyr_1.2.1        tibble_3.1.8       ggplot2_3.3.6     
[17] tidyverse_1.3.2   

loaded via a namespace (and not attached):
  [1] readxl_1.4.1           backports_1.4.1        plyr_1.8.7            
  [4] igraph_1.3.5           splines_4.2.1          usethis_2.1.6         
  [7] GenomeInfoDb_1.32.4    digest_0.6.30          foreach_1.5.2         
 [10] htmltools_0.5.3        fansi_1.0.3            magrittr_2.0.3        
 [13] memoise_2.0.1          googlesheets4_1.0.1    cluster_2.1.4         
 [16] tzdb_0.3.0             remotes_2.4.2          Biostrings_2.64.1     
 [19] modelr_0.1.9           bayesm_3.1-4           vroom_1.5.7           
 [22] prettyunits_1.1.1      colorspace_2.0-3       rvest_1.0.3           
 [25] haven_2.5.1            xfun_0.33              callr_3.7.2           
 [28] crayon_1.5.2           RCurl_1.98-1.8         jsonlite_1.8.0        
 [31] survival_3.4-0         iterators_1.0.14       glue_1.6.2            
 [34] gtable_0.3.1           gargle_1.2.1           zlibbioc_1.42.0       
 [37] XVector_0.36.0         pkgbuild_1.3.1         Rhdf5lib_1.18.2       
 [40] BiocGenerics_0.42.0    DEoptimR_1.0-11        scales_1.2.1          
 [43] DBI_1.1.3              miniUI_0.1.1.1         Rcpp_1.0.9            
 [46] xtable_1.8-4           bit_4.0.4              stats4_4.2.1          
 [49] profvis_0.3.7          htmlwidgets_1.5.4      httr_1.4.4            
 [52] ellipsis_0.3.2         urlchecker_1.0.1       pkgconfig_2.0.3       
 [55] farver_2.1.1           dbplyr_2.2.1           utf8_1.2.2            
 [58] tidyselect_1.2.0       labeling_0.4.2         rlang_1.0.6           
 [61] reshape2_1.4.4         later_1.3.0            munsell_0.5.0         
 [64] cellranger_1.1.0       tools_4.2.1            cachem_1.0.6          
 [67] cli_3.4.1              generics_0.1.3         ade4_1.7-19           
 [70] devtools_2.4.4         broom_1.0.1            evaluate_0.16         
 [73] biomformat_1.24.0      fastmap_1.1.0          yaml_2.3.5            
 [76] processx_3.7.0         knitr_1.40             bit64_4.0.5           
 [79] fs_1.5.2               robustbase_0.95-0      nlme_3.1-159          
 [82] mime_0.12              xml2_1.3.3             compiler_4.2.1        
 [85] rstudioapi_0.14        curl_4.3.2             reprex_2.0.2          
 [88] stringi_1.7.8          ps_1.7.1               Matrix_1.5-1          
 [91] tensorA_0.36.2         multtest_2.52.0        vctrs_0.5.0           
 [94] pillar_1.8.1           lifecycle_1.0.3        rhdf5filters_1.8.0    
 [97] data.table_1.14.2      bitops_1.0-7           httpuv_1.6.6          
[100] R6_2.5.1               promises_1.2.0.1       IRanges_2.30.1        
[103] sessioninfo_1.2.2      codetools_0.2-18       MASS_7.3-58.1         
[106] assertthat_0.2.1       pkgload_1.3.0          rhdf5_2.40.0          
[109] withr_2.5.0            S4Vectors_0.34.0       GenomeInfoDbData_1.2.8
[112] mgcv_1.8-40            parallel_4.2.1         hms_1.1.2             
[115] grid_4.2.1             rmarkdown_2.16         googledrive_2.0.0     
[118] Biobase_2.56.0         shiny_1.7.2            lubridate_1.8.0       </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>